### PR TITLE
Fix CI tests after default authentication mode changed to Scope

### PR DESCRIPTION
**Pull request by ancaantochi**
CI tests were failing because the authentication mode was changed to Scope only. A few changes to the tests to make it work:
- leaf devices should have scope set to the edge device scope
- fixture should use edgeHub sas key to connect to cloud because IotHub doesn't authorize to call scope API with device sas key

### CODE REVIEW

File: edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/ProtocolHeadFixture.cs [hunk 1]
```diff
@@ -78,7 +78,19 @@ async Task StartProtocolHead()
...
+            string iothub = ConnectionStringHelper.GetHostName(edgeDeviceConnectionString);
+            RegistryManager rm = RegistryManager.CreateFromConnectionString(iotHubConnectionString);
+            var edgeHubModule = await rm.GetModuleAsync(edgeDeviceId, "$edgeHub");
+            if (edgeHubModule.Authentication.Type == AuthenticationType.None)
+            {
+                edgeHubModule.Authentication.Type = AuthenticationType.Sas;
+                edgeHubModule = await rm.UpdateModuleAsync(edgeHubModule);
+            }
+
+            ConfigHelper.TestConfig[EdgeHubConstants.ConfigKey.IotHubConnectionString] = $"HostName={iothub};DeviceId={edgeDeviceId};ModuleId=$edgeHub;SharedAccessKey={edgeHubModule.Authentication.SymmetricKey.PrimaryKey}";
```
Comment (and-rewsmith) [inline edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/ProtocolHeadFixture.cs:93]: How is this connection string we are setting different from the ```edgeDeviceConnectionString``` we get from keyvault above?

File: test/modules/CloudToDeviceMessageTester/Settings.cs [hunk 2]
```diff
@@ -54,7 +54,8 @@ class Settings
                     ApiVersion = Preconditions.CheckNonWhiteSpace(apiVersion, nameof(apiVersion)),
                     ModuleGenerationId = Preconditions.CheckNonWhiteSpace(moduleGenerationId, nameof(moduleGenerationId)),
                     IotHubHostName = Preconditions.CheckNonWhiteSpace(iotHubHostName, nameof(iotHubHostName)),
-                    TransportType = transportType
+                    TransportType = transportType,
+                    EdgeDeviceId = deviceId
```
Comment (and-rewsmith) [inline test/modules/CloudToDeviceMessageTester/Settings.cs:58]: nit: both should be edgeDeviceId?

File: edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/ProtocolHeadFixture.cs (hunk 1 above)
Comment (ancaantochi) [reply edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/ProtocolHeadFixture.cs:93]: keyvault has device connection string and here it's using edgeHub connection string

File: test/modules/CloudToDeviceMessageTester/Settings.cs (hunk 2 above)
Comment (ancaantochi) [reply test/modules/CloudToDeviceMessageTester/Settings.cs:58]: deviceId is read from env variable and for leaf device is using that name and appends "-leaf", so deviceId is actually edgeDeviceId

**Review by and-rewsmith — APPROVED** [review]

**Pull request merged**